src/main/scala/org/tikv/bulkload/RawKVBulkLoader.scala

```diff
@@ -0,0 +1,260 @@
+
+  def bulkLoad(rdd: RDD[(Array[Byte], Array[Byte])]): Unit = {
+    tiConf.setKvMode("RAW")
+
+    tiSession = TiSession.create(tiConf)
+
+    // 2 sort
+    val rdd2 = rdd.map { pair =>
+      (SerializableKey(pair._1), pair._2)
+    }.sortByKey().persist()
+
+    // 3 calculate regionSplitPoints
+    val orderedSplitPoints = getRegionSplitPoints(rdd2)
+
+    // 4 switch to normal mode
+    val switchTiKVModeClient = new SwitchTiKVModeClient(tiSession.getPDClient, tiSession.getImporterRegionStoreClientBuilder)
+    switchTiKVModeClient.switchTiKVToNormalMode()
+
+    // 5 call region split and scatter
+    tiSession.splitRegionAndScatter(
+      orderedSplitPoints.map(_.bytes).asJava,
+      optionsSplitRegionBackoffMS,
+      optionsScatterRegionBackoffMS,
+      optionsScatterWaitMS)
+
+    // 7 switch to import mode
+    switchTiKVModeClient.keepTiKVToImportMode()
+
+    // 6 refetch region info
+    val minKey = rdd2.map(p => p._1).min().getRowKey
+    val maxKey = rdd2.map(p => p._1).max().getRowKey
+    val orderedRegions = getRegionInfo(minKey, maxKey)
+    logger.info("orderedRegions size = " + orderedRegions.size)
+
+    //8  repartition rdd according region
+    partitioner = new TiRegionSplitPartitionerV2(orderedRegions)
+    val rdd3 = rdd2.partitionBy(partitioner).persist(StorageLevel.MEMORY_AND_DISK)
+    logger.info("rdd3.getNumPartitions = " + rdd3.getNumPartitions)
+
+    rdd3.foreachPartition { itor =>
+      writeAndIngest(itor.map(pair => (pair._1.bytes, pair._2)), partitioner)
+    }
+    switchTiKVModeClient.stopKeepTiKVToImportMode()
+    switchTiKVModeClient.switchTiKVToNormalMode()
+    logger.info("finish to load data.")
+    tiSession.close()
+  }
+
+  private def writeAndIngest(iterator: Iterator[(Array[Byte], Array[Byte])], partitioner: TiRegionSplitPartitionerV2): Unit = {
+    val (itor1, tiro2) = iterator.duplicate
+
+    var minKey: Key = Key.MAX
+    var maxKey: Key = Key.MIN
+    var region: TiRegion = null
+    var key: Key = null
+    val dataSize = itor1.map { itor =>
+      key = Key.toRawKey(itor._1)
+
+      if (region == null) {
+        region = partitioner.getRegion(key)
+      }
+
+      if (key.compareTo(minKey) < 0) {
+        minKey = key
+      }
+      if (key.compareTo(maxKey) > 0) {
+        maxKey = key
+      }
+    }.size
+
+    if (dataSize > 0) {
+      if (region == null) {
+        logger.warn("region == null, skip ingest this partition")
+      } else {
+        val uuid = genUUID()
+
+        logger.info(s"start to ingest this partition ${util.Arrays.toString(uuid)}")
+        val pairsIterator = tiro2.map { keyValue => new Pair[ByteString, ByteString](ByteString.copyFrom(keyValue._1), ByteString.copyFrom(keyValue._2))
+
+        }.asJava
+        val tiSession = TiSession.create(tiConf)
+        val importerClient = new ImporterClient(tiSession, ByteString.copyFrom(uuid), minKey, maxKey, region, ttl)
+        importerClient.rawWrite(pairsIterator)
```

Comment:
```suggestion
        importerClient.write(pairsIterator)
```

Reply:
API changes https://github.com/tikv/client-java/pull/254

Reply:
Fixed in commit [[BulkLoad: fix README.md and keys must be added in order exception](https://github.com/tikv/migration/pull/1/commits/3c0520a1d4aefee81148474a77b47ff0fcacc6da)]

Reply:
@liangyouze you can click "Resolve conversation" if the comment has been addressed.

Reply:
![image](https://user-images.githubusercontent.com/18818196/130554365-5d877dbf-5b38-4c99-8e2f-adf2a05aa8da.png)